.github/workflows/python-validate-test.yml

```
@@ -10,29 +10,24 @@ on:
     branches: [ master ]
 
 jobs:
-  test:
-    runs-on: ubuntu-20.04
+  test-legacy:
+    runs-on: ubuntu-24.04
     env:
       # Temporary workaround for Python 3.5 failures - May 2024, see CFE-4395
       PIP_TRUSTED_HOST: "pypi.python.org pypi.org files.pythonhosted.org"
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.5]
+        python-version: ["3.5.10"]
 
     steps:
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+    - name: Set up legacy Python ${{ matrix.python-version }}
+      uses: ./.github/actions/set-up-legacy-python
```

I don't think the dot-slash is necessary here :thinking: 
```suggestion
      uses: .github/actions/set-up-legacy-python
```